packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx

```diff
@@ -169,6 +266,135 @@ export const ReferencePropertySection = ({
       }
     }
   }, [clickLoading, clickError, clickData]);
+  useEffect(() => {
+    if (!clearLoading) {
+      if (clearError) {
+        addErrorMessage('An unexpected error has occurred, please refresh the page');
+      }
+      if (clearData) {
+        const { clearReference } = clearData;
+        if (isErrorPayload(clearReference) || isSuccessPayload(clearReference)) {
+          addMessages(clearReference.messages);
+        }
+      }
+    }
+  }, [clearLoading, clearError, clearData]);
+  useEffect(() => {
+    if (!removeLoading) {
+      if (removeError) {
+        addErrorMessage('An unexpected error has occurred, please refresh the page');
+      }
+      if (removeData) {
+        const { removeReferenceValue } = removeData;
+        if (isErrorPayload(removeReferenceValue) || isSuccessPayload(removeReferenceValue)) {
```

Comment:
Did I missed something or is there an issue here. You method `isErrorPayload` is defined as:

```
const isErrorPayload = (payload: GQLClickReferenceValuePayload): payload is GQLErrorPayload => payload.__typename === 'ErrorPayload';
```

but the payload here is definitely not a `GQLClickReferenceValudPayload`. Either I've missed something or your type hierarchy is not correct.

Reply:
Changed